### PR TITLE
3.0: PCluster components refactoring

### PR DESCRIPTION
**Pull request by lukeseawalker**
**Please See** [Git Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions)

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cloudformation/imagebuilder/pcluster_install.yaml [hunk 1]
```diff
@@ -196,7 +239,7 @@ phases:
               set -x
               chef-client --local-mode --config /etc/chef/client.rb --log_level info --logfile /var/log/chef-client.log --force-formatter --no-color --chef-zero-port 8889 --json-attributes /tmp/dna.json --override-runlist aws-parallelcluster::default
 
-      - name: Bootstrap
+      - name: InstallCfnBootstrap
         action: ExecuteBash
         inputs:
           commands:
```
Comment (yuleiwan) [inline cloudformation/imagebuilder/pcluster_install.yaml:245]: use separate action for bucket url in bootstrap

File: cloudformation/imagebuilder/pcluster_install.yaml [hunk 2]
```diff
@@ -133,33 +174,35 @@ phases:
...
       - name: DownloadCookbook
         action: ExecuteBash
         inputs:
           commands:
             - |
               set -x
-              CookbookVersion="aws-parallelcluster-cookbook-2.10.1"
-              cookbook_url=https://${AWS::Region}-aws-parallelcluster.s3.${AWS::Region}.amazonaws.com/cookbooks/${!CookbookVersion}.tgz
-              echo "${!cookbook_url}"
+              # TODO move vars out in separate actions
```
Comment (yuleiwan) [inline cloudformation/imagebuilder/pcluster_install.yaml:184]: Do you want to address the TODO in this patch?

**Review by yuleiwan — APPROVED** [review]
LGTM

File: cloudformation/imagebuilder/pcluster_install.yaml (hunk 1 above)
Comment (lukeseawalker) [reply cloudformation/imagebuilder/pcluster_install.yaml:262]: done

File: cloudformation/imagebuilder/pcluster_install.yaml (hunk 2 above)
Comment (lukeseawalker) [reply cloudformation/imagebuilder/pcluster_install.yaml]: done

**Review by yuleiwan — APPROVED** [review]

**Pull request merged**